gapic/src/main/com/google/gapid/widgets/LoadableImage.java

```diff
@@ -235,6 +176,82 @@ public default void onLoadingStart() { /* empty */ }
     public default void onLoaded(boolean success) { /* empty */ }
   }
 
+  /**
+   * Builder for {@link LoadableImage}. If built using a future, it is assumed
+   * to already be loading, while if built with a supplier, the {@link #load()}
+   * method needs to be invoked to start the loading process.
+   */
+  public static class Builder {
+    private final LoadingIndicator loading;
+    private Supplier<ListenableFuture<Object>> futureSupplier;
+    private boolean small;
+    private boolean shouldLoad;
```

Comment:
`shouldLoad` is a bit magic. Perhaps some documentation explaining what it is and why would help someone like me understand why it exists.